spinalcordtoolbox/scripts/sct_label_vertebrae.py

```diff
@@ -250,27 +242,31 @@ def main(argv=None):
         fname_disc = os.path.abspath(arguments.discfile)
     else:
         fname_disc = None
-    if arguments.initz is not None:
-        initz = arguments.initz
-        if len(initz) != 2:
-            raise ValueError('--initz takes two arguments: position in superior-inferior direction, label value')
-    if arguments.initcenter is not None:
-        initcenter = arguments.initcenter
+
+    initz = arguments.initz
+    initcenter = arguments.initcenter
     # if user provided text file, parse and overwrite arguments
     if arguments.initfile is not None:
-        file = open(arguments.initfile, 'r')
-        initfile = ' ' + file.read().replace('\n', '')
-        arg_initfile = initfile.split(' ')
-        for idx_arg, arg in enumerate(arg_initfile):
+        args = open(arguments.initfile).read().split()
+        iterator = iter(args)
+        for arg in iterator:
             if arg == '-initz':
-                initz = [int(x) for x in arg_initfile[idx_arg + 1].split(',')]
-                if len(initz) != 2:
-                    raise ValueError('--initz takes two arguments: position in superior-inferior direction, label value')
-            if arg == '-initcenter':
-                initcenter = int(arg_initfile[idx_arg + 1])
+                try:
+                    initz = [int(x) for x in next(iterator).split(',')]
+                except (StopIteration, ValueError):
+                    parser.error('-initz takes two arguments: position in superior-inferior direction, label value')
+            elif arg == '-initcenter':
+                try:
+                    initcenter = int(next(iterator))
+                except (StopIteration, ValueError):
+                    parser.error('-initcenter takes an integer argument')
+    if initz is not None and len(initz) != 2:
+        parser.error('-initz takes two arguments: position in superior-inferior direction, label value')
+
```

Comment:
It could be handled by `argparse` using a custom type function, but not with `nargs=2`. The reason being, we accept the syntax `-initz 67,3` (with a comma) whereas `nargs=2` would accept the syntax `-initz 67 3` (with a space). But I'm very open to the idea of using a custom type function, as we did for `-param` in #3700.